src/pybind/mgr/volumes/fs/volume.py

```diff
@@ -367,6 +367,12 @@ def conn_wrapper(self, fs_handle, **kwargs):
             return result
         return conn_wrapper
 
+    def nametojson(self, names):
+        dict = []
+        for i in range(len(names)):
+            dict.append({'name': names[i]})
```

Comment:
Should be names[i].decode('utf-8'). To convert to string object.

Reply:
Again you're trying to json.dump a bytes object. This doesn't work. Please *test with py3 * environment before you push.

This is the error I get.
```
  File "/home/rraja/git/ceph/src/pybind/mgr/volumes/fs/volume.py", line 459, in list_subvolumes
    ret = self.nametojson(subvolumes)
  File "/home/rraja/git/ceph/src/pybind/mgr/volumes/fs/volume.py", line 374, in nametojson
    return 0, json.dumps(dict, indent=2), ""
  File "/usr/lib64/python3.7/json/__init__.py", line 238, in dumps
    **kw).encode(obj)
  File "/usr/lib64/python3.7/json/encoder.py", line 201, in encode
    chunks = list(chunks)
  File "/usr/lib64/python3.7/json/encoder.py", line 429, in _iterencode
    yield from _iterencode_list(o, _current_indent_level)
  File "/usr/lib64/python3.7/json/encoder.py", line 325, in _iterencode_list
    yield from chunks
  File "/usr/lib64/python3.7/json/encoder.py", line 405, in _iterencode_dict
    yield from chunks
  File "/usr/lib64/python3.7/json/encoder.py", line 438, in _iterencode
    o = _default(o)
  File "/usr/lib64/python3.7/json/encoder.py", line 179, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type bytes is not JSON serializable


```

Reply:
I have decoded the names now. But you are testing the older code.

> ret = self.nametojson(subvolumes)

This doesn't exist in volume.py.